rllib/execution/parallel_requests.py

```diff
@@ -178,7 +180,237 @@ def wait_asynchronous_requests(
+            )
+        if fn_args is None:
+            fn_args = []
+        if fn_kwargs is None:
+            fn_kwargs = {}
+        call = (remote_fn, fn_args, fn_kwargs)
+        if actor:
+            if actor not in self._all_workers:
+                raise ValueError(
+                    f"Actor {actor} has not been added to the manager."
+                    f" You must call manager.add_worker(actor) first "
+                    f"before submitting requests to actor."
+                )
+            self._call_queue[actor].put(call)
+        elif for_all_workers:
+            for actor in self._all_workers:
+                self._call_queue[actor].put(call)
+        else:  # Submit to a random worker.
+            if len(self._all_workers) == 0:
+                raise ValueError("No workers available to submit request.")
+            elif len(self._all_workers) == 1:
+                actor = list(self._all_workers)[0]
+            else:
+                # load balance by getting 2 random actors and then placing the call
+                # on the actor with the least number of inflight requests. If max
+                # inflight requests is reached, place the call on the actor with the
+                # smaller queue
+                [a1, a2] = random.sample(self._all_workers, 2)
+                num_inflight_req_a1 = len(self._remote_requests_in_flight[a1])
+                num_inflight_req_a2 = len(self._remote_requests_in_flight[a2])
+                if num_inflight_req_a1 < num_inflight_req_a2:
+                    actor = a1
+                elif num_inflight_req_a1 > num_inflight_req_a2:
+                    actor = a2
+                else:
+                    if self._call_queue[a1].qsize() < self._call_queue[a2].qsize():
+                        actor = a1
+                    else:
+                        actor = a2
+            self._call_queue[actor].put(call)
+
+        self._run()
+
+    def _run(self) -> int:
+        """Launch the submited requests remotely on workers that don't have more
+        than max_inflight_requests in flight.
+        """
+        num_requests_launched = 0
+        for actor, q in self._call_queue.items():
+            # print(len(self._remote_requests_in_flight[actor]))
+            # print(self._max_remote_requests_in_flight)
+            while (
+                len(self._remote_requests_in_flight[actor])
+                < self._max_remote_requests_in_flight
+            ) and q.qsize() > 0:
+                remote_fn, fn_args, fn_kwargs = q.get()
+                req = actor.apply.remote(remote_fn, *fn_args, **fn_kwargs)
+                self._remote_requests_in_flight[actor].add(req)
+                self._pending_to_actor[req] = actor
+                self._pending_remotes.append(req)
+                num_requests_launched += 1
+        return num_requests_launched
+
+    def get_ready_results(self) -> Dict[ActorHandle, List[Any]]:
```

Comment:
thinking about the APIs a bit more, I feel iike it may be better to name them:

AsyncManager.call(actor, fn). # actor may be None, in which case we call a random actor.
AsyncManager.call_all(fn)
AsyncManager.get_ready()

wdyt?